crates/paimon/src/spec/snapshot.rs

```diff
@@ -135,3 +135,148 @@ impl Snapshot {
+        assert_eq!(snapshot.delta_manifest_list(), "delta_manifest_list_v2");
+        assert_eq!(
+            snapshot.changelog_manifest_list(),
+            Some("changelog_manifest_list_v2")
+        );
+        assert_eq!(snapshot.index_manifest(), Some("index_manifest_list_v2"));
+        assert_eq!(snapshot.commit_user(), "user2");
+        assert_eq!(snapshot.total_record_count(), Some(500));
+        assert_eq!(snapshot.delta_record_count(), Some(200));
+        assert_eq!(snapshot.changelog_record_count(), Some(50));
+        assert_eq!(snapshot.watermark(), Some(123456789));
+        assert_eq!(snapshot.statistics(), Some("statistics_v2"));
+    }
+
+    #[test]
+    fn test_snapshot_with_none_fields() {
+        let snapshot = Snapshot::builder()
+            .version(3)
+            .id(1003)
+            .schema_id(2004)
+            .base_manifest_list("base_manifest_list_v3".to_string())
+            .delta_manifest_list("delta_manifest_list_v3".to_string())
+            .changelog_manifest_list(None)
+            .index_manifest(None)
+            .commit_user("user3".to_string())
+            .total_record_count(None)
+            .delta_record_count(None)
+            .changelog_record_count(None)
+            .watermark(None)
+            .statistics(None)
+            .build();
+
+        assert_eq!(snapshot.version(), 3);
+        assert_eq!(snapshot.id(), 1003);
+        assert_eq!(snapshot.schema_id(), 2004);
+        assert_eq!(snapshot.base_manifest_list(), "base_manifest_list_v3");
+        assert_eq!(snapshot.delta_manifest_list(), "delta_manifest_list_v3");
+        assert_eq!(snapshot.commit_user(), "user3");
+        assert!(snapshot.changelog_manifest_list().is_none());
+        assert!(snapshot.index_manifest().is_none());
+        assert!(snapshot.total_record_count().is_none());
+        assert!(snapshot.delta_record_count().is_none());
+        assert!(snapshot.changelog_record_count().is_none());
+        assert!(snapshot.watermark().is_none());
+        assert!(snapshot.statistics().is_none());
+    }
+
+    #[test]
+    fn test_snapshot_serialization_deserialization() {
+        let original_snapshot = Snapshot::builder()
+            .version(4)
+            .id(1004)
+            .schema_id(2005)
+            .base_manifest_list("base_manifest_list.json".to_string())
+            .delta_manifest_list("delta_manifest_list.json".to_string())
+            .changelog_manifest_list(Some("changelog_manifest_list.json".to_string()))
+            .index_manifest(Some("index_manifest.json".to_string()))
+            .commit_user("test_user".to_string())
+            .total_record_count(Some(1000))
+            .delta_record_count(Some(100))
+            .changelog_record_count(Some(50))
+            .watermark(Some(1234567890))
+            .statistics(Some("statistics.json".to_string()))
+            .build();
+
+        let serialized =
```

Comment:
Hi, sorry for the confusion. I'm referring to testing against actual JSON data like this:

The following json is generated by a paimon-java:

```json
{
  "version" : 3,
  "id" : 1,
  "schemaId" : 0,
  "baseManifestList" : "manifest-list-4091b92b-01d3-4b91-8e47-8c9f61847d2f-0",
  "deltaManifestList" : "manifest-list-4091b92b-01d3-4b91-8e47-8c9f61847d2f-1",
  "changelogManifestList" : null,
  "indexManifest" : "index-manifest-55e3e815-08ab-4a70-a808-a8df5d275cb4-0",
  "commitUser" : "cf568e07-05ad-4943-b4bd-37461bc58729",
  "commitIdentifier" : 9223372036854775807,
  "commitKind" : "APPEND",
  "timeMillis" : 1721287833568,
  "logOffsets" : { },
  "totalRecordCount" : 1,
  "deltaRecordCount" : 1,
  "changelogRecordCount" : 0,
  "watermark" : -9223372036854775808
}
```

We can test our snapshot implementation by serving it through this method.